### ISSUE TITLE
Last error report

### ISSUE THREAD

**Issue by leo-beliakov** [getsentry/sentry-java#3288]
### Problem Statement

With AppCenter we were able to retrieve the last crash report. It seems not possible to do with Sentry. 

Here's a sample code for AppCenter: 
```
if(Crashes.hasCrashedInLastSession()){
            val report = Crashes.getLastSessionCrashReport()
            // work with the last report
}
```

**Comment by markushi** [comment]
Hi @leo-beliakov , thanks for reaching out! Could you explain your use case a bit more? 
What would do with the last report?

A few pointers which could be helpful:
1. We have a `beforeSend` hook, allowing you to get notified or even modify events, including crashes
2. We have `Sentry.isCrashedLastRun()` which tells you if the last app run crashed

**Comment by leo-beliakov** [comment]
@markushi the use case is simple: we are using Mixpanel for analytics and AppCenter for crash reporting. But we want to track occurred crashes in Mixpanel, too. AppCenter allows us to get the information about the last crash via `getLastSessionCrashReport()` while in SentryAndroid I don't see this option (interestingly, on IOS Sentry has `onCrashedLastRun` callback)

**Comment by leo-beliakov** [comment]
Yeah, I found that way with `beforeSend` which looks a bit like a workaround but it seems to be working anyway.

**Comment by leo-beliakov** [comment]
I believe the issue can be closed, but it would be really nice to see more consistency between SDK for Android & IOS

**Comment by stefanosiano** [comment]
On Android we send the event on the crashing run or, if we fail, in the next one, whereas the iOS SDK sends it always on the next run.
This is why the implementation is different and iOS has the callback.
We could add a callback, but that would basically be the workaround using the `beforeSend`.
Closing this issue at the moment. We can reevaluate it later
